### PR TITLE
docs(bazel): improving docs on using Bazel

### PR DESCRIPTION
**Pull request by ghost**
### Summary

This PR aims to optimize Kong's documentation on using Bazel, mainly focusing on:

1. explaining the effects of commands.
2. adding necessary link documents.
3. optimizing the document structure to make it more clear.
4. adding some debugging tricks for Bazel.

I want to know if this is currently necessary. Or do you have any other suggestions?

### Checklist

- [ ] The Pull Request has tests
- [ ] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* [Implement ...]

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

File: build/README.md [hunk 1]
```diff
@@ -8,34 +8,50 @@ The build system is tested on Linux (x86_64 and arm64) and macOS (Intel chip and
...
 ## Building
 
-To build Kong and all its dependencies, run the following command:
+### Build Kong and its dependencies
+
+The commands related to building Kong are already supported in the [Makefile](../Makefile), so we can use them directly:
 
 ```bash
-bazel build //build:kong --verbose_failures
+make build-kong
```
Comment (fffonion) [inline build/README.md:37]: maybe let's still keep the `bazel build //build:kong --verbose_failures` command here as a `equivelant` option

File: build/README.md [hunk 2]
```diff
@@ -8,34 +8,50 @@ The build system is tested on Linux (x86_64 and arm64) and macOS (Intel chip and
...
+
+Once the build is complete, you will see four `bazel-*` folders in the current directory. Refer to the [workspace layout diagram](https://bazel.build/remote/output-directories?hl=en#layout-diagram) for their respective definitions.
+
+### Build a virtual development environment
 
-To use the build as a virtual development environment, run:
-  
 ```bash
-bazel build //build:venv --verbose_failures
+make build-venv
```
Comment (fffonion) [inline build/README.md:47]: same here

File: build/README.md [hunk 3]
```diff
@@ -8,34 +8,50 @@ The build system is tested on Linux (x86_64 and arm64) and macOS (Intel chip and
...
 The build system requires the following tools to be installed:
 
 - [Bazel/Bazelisk](https://bazel.build/install/bazelisk), Bazelisk is recommended to ensure the correct version of Bazel is used.
-    - Use `make check-bazel` to install Bazelisk into `bin/bazel`, then use `export PATH=bin:$PATH` to add it into your `PATH`.
-- [Build dependencies](https://github.com/Kong/kong/blob/master/DEVELOPER.md#build-and-install-from-source)
 
+  We can install Bazelisk by running the following command:
+
+  ```bash
+    // install Bazelisk into $PWD/bin/bazel
```
Comment (chronolaw) [inline build/README.md:20]:
```suggestion
    # install Bazelisk into $PWD/bin/bazel
```
Using `#` as a comment may be better?

**Review by fffonion — APPROVED** [review]

**Comment by fffonion** [comment]
lgtm, @sabertobihwy i see a WIP in the title, is there anything more you want to add to this PR?

**Comment by sabertobihwy** [comment]
> lgtm, @sabertobihwy i see a WIP in the title, is there anything more you want to add to this PR?

Thank you for pointing that out. For this PR, I think my current changes have completed my initial vision.

**Pull request merged**